lib/actions/appleMusic.js

```diff
@@ -161,9 +173,24 @@ function appleMusic(player, values) {
                 const metadataID = METADATA_URI_STARTERS[type] + encodeURIComponent('song:' + tracksJson.results[trkNum].trackId);
                 const metadata = getMetadata(metadataID, PARENTS[type] + tracksJson.results[trkNum].trackName.toLowerCase(), CLASSES[type],tracksJson.results[trkNum].trackName);
                 const uri = uriTemplates[type](encodeURIComponent('song:' + tracksJson.results[trkNum].trackId));
-                nextTrackNo = player.coordinator.state.trackNo + 1;
-                return player.coordinator.addURIToQueue(uri, metadata, true, nextTrackNo)
-                  .then(() => player.coordinator.trackSeek(nextTrackNo))
+                var   empty = false;
+
+                return player.coordinator.getQueue(0,1)
+                  .then((queue) => {
+                    empty = (queue.numberReturned == 0);
+                    nextTrackNo = (empty)?1:player.coordinator.state.trackNo + 1;
+                  })
+                  .then(() => player.coordinator.addURIToQueue(uri, metadata, true, nextTrackNo))
+                  .then(() => {
+                    if (radioOn(player)) {
+                      player.coordinator.setAVTransport(queueURI,'');
+                    }
+                  })
+                  .then(() => {
+                    if (!empty) {
+                      player.coordinator.nextTrack();
+                    }
```

Comment:
Same thing with setAVTransport and nextTrack, you need to return the promise to make it wait for that to finish. It is preferable to always wait since the players easily choke if it receives multiple calls in parallell.